test/test_blocks.py

```diff
@@ -463,6 +466,115 @@ def test_raise_error_if_event_queued_but_queue_not_enabled(self):
         demo.close()
 
 
+class TestTempFile:
+    def test_pil_images_hashed(self, tmp_path, connect):
+        def create_images(n_images):
+            a = Image.new("RGB", (512, 512), "red")
+            b = Image.new("RGB", (512, 512), "green")
+            c = Image.new("RGB", (512, 512), "blue")
+
+            res = [a, b, c][:n_images]
+            random.shuffle(res)
+            return res
```

Comment:
> I think it's ok for now in this simple test

I can almost bet money that no one will ever touch this test again, so doing it more efficiently now pays off.

Also, you can use `random.sample()`, so...

```suggestion
                images = [Image.new("RGB", (512, 512), color) for color in ("red", "green", "blue")]
                def create_images(n_images):
                    return random.sample(images, n_images)
```

Reply:
The difference is in microseconds but made the switch. The new code _is_ nicer!